libs/pilight/protocols/433.92/quigg_gt9000.c

```diff
@@ -0,0 +1,241 @@
+/*
+	Copyright (C) 2015 dominikkarall
+
+	This file is part of pilight.
+
+	pilight is free software: you can redistribute it and/or modify it under the
+	terms of the GNU General Public License as published by the Free Software
+	Foundation, either version 3 of the License, or (at your option) any later
+	version.
+
+	pilight is distributed in the hope that it will be useful, but WITHOUT ANY
+	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with pilight. If not, see	<http://www.gnu.org/licenses/>
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "../../core/pilight.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/log.h"
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "quigg_gt9000.h"
+
+#define PULSE_QUIGG_SHORT	500
+#define PULSE_QUIGG_LONG	1000
+#define PULSE_QUIGG_FOOTER1	3000
+#define PULSE_QUIGG_FOOTER2	7000
+
+#define NORMAL_REPEATS		4
+#define AVG_PULSE_LENGTH	750
+#define RAW_LENGTH		50
+
+static int validate(void) {
+	if(quigg_gt9000->rawlen == RAW_LENGTH) {
+		if(quigg_gt9000->raw[quigg_gt9000->rawlen-1] >= (int)(PULSE_QUIGG_FOOTER2*0.9) &&
+			 quigg_gt9000->raw[quigg_gt9000->rawlen-1] <= (int)(PULSE_QUIGG_FOOTER2*1.1) &&
+			 quigg_gt9000->raw[quigg_gt9000->rawlen-2] >= (int)(PULSE_QUIGG_FOOTER1*0.9) &&
+			 quigg_gt9000->raw[quigg_gt9000->rawlen-2] <= (int)(PULSE_QUIGG_FOOTER1*1.1)) { 
+		return 0;
+		}
+	}
+	return -1;
+}
+
+static void createMessage(int id, int payload, int unit, char *binary) {
+	quigg_gt9000->message = json_mkobject();
+	if(binary!=0)
```

Comment:
Can you follow pilight coding styles here. Spaces in the if and using accolades.